features/timeseries/src/main/java/org/opennms/netmgt/timeseries/util/TimeseriesUtils.java

```diff
@@ -137,4 +112,11 @@ public static Sample createSampleForIndexingStrings(final String resourceId, Map
                 .build();
     }
 
+    public static String toSearchRegex(ResourcePath path, int depth) {
+        return "^" + // start string
+                toResourceId(path) + // exact match
+                ":[^.:]*".repeat(depth) + // colon (:) plus any chars except colon (:), repeated 'depth' times
```

Comment:
There's a slight edge case here where the `:` character is part of the path. The `EscapableResourceIdSplitter` handles this by prefixing these with `\` so they are "escaped" accordingly" and not treated as separators when parsing.

Is it possible to add test cases and adjust the regex for these? i.e. don't consider a `:` if it is prefixed with a `\`

Reply:
Yeah, that makes sense...
Although thinking of it: Do we need the conversion from "/" to ":" at all? I think we did that for Newts (and thus inherited it in the integration layer) but we might take some complexity out if we remove that transformation? It might be ok to use the path forward slashes?
Do you remember why we have the colons at all?

Reply:
I replaced the colon with a forward slash as discussed. The assumption is that a path can never have a forward slash in an element.